packages/frontend/src/components/create-campaign/form/index.tsx

```diff
@@ -133,16 +138,16 @@ export function CreateCampaignForm({ target }: CreateCampaignFormProps) {
 
     return (
         <div className={styles.root}>
-            {target === TargetType.AmmPoolLiquidity && (
+            {isTargeting(payload, TargetType.AmmPoolLiquidity) && (
```

Comment:
Can't we make this part of a class. Like we do for the `Campaign` and `CampaignPreviewSpecification` type

Reply:
The issue here is that the payload is a state in the component and having a class would mean having an instance in the state and it's probably not the best, so I added that utility.

Reply:
I honestly think it's alright, but we should probably implement setters so that a new instance is returned when an attribute changes. So that React can correctly handle rerenders. I guess we could try, maybe.